src/OpenTelemetry.Extensions/Trace/AutoFlushActivityProcessor.cs

```diff
@@ -0,0 +1,60 @@
+﻿// <copyright file="AutoFlushActivityProcessor.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System;
+using System.Diagnostics;
+using System.Threading;
+
+namespace OpenTelemetry.Trace
+{
+    /// <summary>
+    /// Activity processor that flushes its containing <see cref="TracerProvider"/> if an ended
+    /// activity matches a predicate.
```

Comment:
What if the predicate triggered new Activity creation?

Reply:
It's assumed that predicate is defined as a logical expression and doesn't contain more complex code. Predicate triggering new Activity creation seems more exotic scenario.

Reply:
Consider something like https://github.com/open-telemetry/opentelemetry-dotnet/blob/cab5b26a24f666cca6089e380adb83805a02eee4/src/OpenTelemetry/Metrics/MetricReader.cs#L116-L124

Reply:
Do you mean only comment from the reference or also the code preventing parallel calls to the `OnCollect(...)` method ?